evm/src/fixed_recursive_verifier.rs

```diff
@@ -1145,13 +1203,26 @@ where
         block_inputs
             .set_verifier_data_target(&self.block.cyclic_vk, &self.block.circuit.verifier_only);
 
-        set_public_value_targets(&mut block_inputs, &self.block.public_values, &public_values)
-            .map_err(|_| {
-                anyhow::Error::msg("Invalid conversion when setting public values targets.")
-            })?;
+        // This is basically identical to this block public values, apart from the `trie_roots_before`
+        // that may come from the previous proof, if any.
```

Comment:
Shouldn't only the state trie root come from the previous proof? Aren't the other tries initialized at the beginning of each block?